prqlc/prql-compiler/src/semantic/resolver/transforms.rs

```diff
@@ -395,44 +397,35 @@ pub fn resolve_special_func(resolver: &mut Resolver, closure: Func) -> Result<Ex
     let transform_call = TransformCall {
         kind: Box::new(kind),
         input: Box::new(input),
-        partition: Vec::new(),
+        partition: None,
         frame: WindowFrame::default(),
         sort: Vec::new(),
     };
     Ok(Expr::new(ExprKind::TransformCall(transform_call)))
 }
 
-/// Wraps non-tuple Exprs into a singleton Tuple.
-// This function should eventually be applied to all function arguments that
-// expect a tuple.
-pub fn coerce_into_tuple(expr: Expr) -> Result<Vec<Expr>> {
-    Ok(match expr.kind {
-        ExprKind::Tuple(items) => {
+impl Resolver<'_> {
+    /// Wraps non-tuple Exprs into a singleton Tuple.
+    pub(super) fn coerce_into_tuple(&mut self, expr: Expr) -> Result<Expr> {
+        let is_tuple_ty = expr.ty.as_ref().unwrap().is_tuple() && !(expr.kind.is_all());
+        Ok(if is_tuple_ty {
+            // a helpful check for a common mis-pattern
```

Comment:
weirdly `typos` complains about `mis`! So I changed

```suggestion
            // a helpful check for a common anti-pattern
```